src/components/hoc/withMeeting.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {concatMap} from 'rxjs/operators';
+import PropTypes from 'prop-types';
+import {AdapterContext} from '../hooks/contexts';
+
+/**
+ * withMeeting is a higher-order component that takes a component (WrappedComponent)
+ * and returns a new component (WithMeeting). The new component needs to be provided with a meeting destination
+ * on the "meetingDestination" prop. The wrapped component will receive the meeting object on the "meeting" prop.
+ *
+ * Example:
+ *
+ * class OriginalComponent extends Component {
+ *   render(props) {
+ *     return <div>Meeting id is {props.meeting.ID}</div>;
+ *   }
+ * }
+ *
+ * const EnhancedComponent = withMeeting(OriginalComponent);
+ *
+ * <EnhancedComponent meetingDestination="<MEETING_DESTINATION>" />
+ *
+ * @param {React.Component} WrappedComponent The component to wrap and enhance
+ * @returns {React.Component} The wrapped component
+ */
+
+/**
+ * @param WrappedComponent
+ */
+export default function withMeeting(WrappedComponent) {
+  class WithMeeting extends React.Component {
```

Comment:
Even though the usage for this is to use it with class components, does the HOC have to use class style? could it be a function component like the rest of the code base?

Reply:
Hi @lalli-flores Yes, it is possible to create a functional component instead of class component. I will make an amend commit with the functional component. What's your opinion?

Reply:
I would prefer that @lungancatalin, to keep the code base consistent. Thanks!

Reply:
Hi @lalli-flores, the commit with functional component is already done. Please have a look on this commit -> https://github.com/webex/components/commit/1418217c480c9da90d0738f2f2cacd4ced920f8c
Regards